pyproject.toml

```diff
@@ -0,0 +1,38 @@
+[tool.poetry]
+name = "klaam"
+version = "0.0.1"
+description = "Arabic speech recognition, classification and text-to-speech."
+license = "MIT"
+authors = [
+    "Zaid Alyafeai <alyafey22@gmail.com>",
+    "mustafa0x"
+]
+readme = 'README.md'
+
+[tool.poetry.dependencies]
+python = "^3.8"
+rich = "^12.2.0"
+jupyter = "^1.0.0"
+jiwer = "^2.3.0"
+arabic-pronounce = "^0.2.5"
+mishkal = "^0.4.1"
+codernitydb3 = "^0.6.0"
+pypinyin = "^0.46.0"
+transformers = "^4.18.0"
```

Comment:
I am not much familiar with poetry and pyproject.toml syntax. However, if '^' means this version or higher, then I do not think this is recommended. This is because backward compatibility is not guaranteed in such highly dynamic packages. I run into these issues before, one of these issues was with jiwer package :(.

Reply:
You are correct `^` means this version or higher, however, when it installs it will only install that exact version, not the latest version possible (to my knowledge). However, what this tag is helpful is when you have a new package that you want to install but it requires a high version of your already installed package, so poetry would install the new package and update the old package with a new version.

For example, I have installed numpy and got this version `numpy = "^0.1.0"`, then I wanted to install `pandas` which requires `numpy >= 0.5.0` so `poetry` will update numpy to be `numpy = "^0.5.0"` and then install `pandas`.

You can restrict the versioning of packages in two ways:
- When installing the package, explicitly state the version of the package. For example, `poetry add numpy=0.1.0` will install numpy with `0.1.0` then append/update `pyproject.toml` to be `numpy = "0.1.0"`.
- Editing `pyproject.toml` to remove `^` from the packages then reinstall the packages.

Both methods will install a specific version of `numpy` but for any new library that requires different version, `poetry` will scream at you to tell you that your project has `numpy=0.1.0` but the newer packages require a different version of `numpy`.

Reply:
I think we should stick with your approach @sudomaze . Thanks for the details.